apps/settings/js/panels/addons/addons_template.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * The template function for generating an UI element for an Addon item.
+ *
+ * @module addons/addons_template
+ */
+define(function(require) {
+  'use strict';
+
+  return function addonsTemplate(item, recycled) {
+    var manifest = item.manifest;
+    var container = null;
+    var span, toggle;
+    if (recycled) {
+      container = recycled;
+      toggle = container.querySelector('input');
+      span = container.querySelector('span');
+    } else {
+      var kind = 'checkbox';
+      container = document.createElement('li');
+      toggle = document.createElement('input');
+      var label = document.createElement('label');
+      span = document.createElement('span');
+      label.className = 'pack-' + kind;
+      toggle.type = kind;
+      toggle.value = item.manifestURL;
+      toggle.checked = item.enabled;
+      label.appendChild(toggle);
+      label.appendChild(span);
+      container.appendChild(label);
+    }
+
+    span.textContent = manifest.name;
+    toggle.onclick = () => {
```

Comment:
We could make the handler passed from `addons_list.js` and then we can call the method of the manager there without attaching the manager to the app instance.